dns-controller/cmd/dns-controller/main.go

```diff
@@ -74,6 +79,11 @@ func main() {
 	flags.AddGoFlagSet(flag.CommandLine)
 	flags.Parse(os.Args)
 
+	go func() {
+		http.Handle("/metrics", promhttp.Handler())
```

Comment:
My only concern is that we have no way to shut this new feature off.  That capability adds more complexity to this PR, but it we probably need the capability to control that functionality.

@robinpercy thoughts?

Reply:
@chrislovecnm  It's not a particularly invasive feature. But after looking at how we do it elsewhere (eg calico), I think we should expose it and allow it to be disabled via the API, for consistency.

@tomwilkie We could add two variables to the ExternalDNS API, one to toggle the feature, and the other to override the port. It would be similar to how the WatchIngress flag was added in https://github.com/kubernetes/kops/pull/2504/

Let us know if you don't have time to add that in, and maybe we can merge this as-is and do the API stuff in another PR

Reply:
Yeah no problem, will do it tomorrow.

> and the other to override the port

Port is configurable via `-metrics-listen-addr`.

Reply:
Great thanks. 

Since kops generates the manifest for dns-controller at cluster-build time, we need to add a field for `-metrics-listen-addr` to the internal model and API in order for users to be able to specify it. Unfortunately, there are a few hoops to jump through :/. Let me know if you have any questions. #2504 is the best example I could find.